File: Languages/Go/Day9.go
// Languages - Go - Day 9
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a program that uses a loop to print the numbers from 1 to 10 and calculates their sum.

package main

import "fmt"

func calculateSumInRange(start int, end int) {
	sum := 0
	for i := start; i <= end; i++ {
		fmt.Println(i)
		sum += i
	}
	fmt.Printf("The sum of the numbers from %d to %d is: %d\n", start, end, sum)
}

func main() {
	calculateSumInRange(1, 10)
}

File: Languages/R/Day9.r
# Languages - R - Day 9
# Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

# Create a program that uses a loop to print the numbers from 1 to 10 and calculates their sum.

calculate_sum_in_range <- function(start_range, end_range) {
  sum <- 0
  for (i in start_range:end_range) {
    print(i)
    sum <- sum + i
  }
  print(paste("The sum of the numbers from", start_range, "to", end_range, "is:", sum))
}

calculate_sum_in_range(1, 10)
